packages/phone-number-privacy/src/index.ts

```diff
@@ -1,10 +1,17 @@
 import { BLINDBLS } from 'bls12377js-blind'
 import * as functions from 'firebase-functions'
-import { PHONE_NUMBER_PRIVACY_SECRET_KEY } from './config'
+import config, { connectToDatabase } from './config'
 
 export const getSalt = functions.https.onRequest((request, response) => {
-  const privateKey = new Buffer(PHONE_NUMBER_PRIVACY_SECRET_KEY)
+  const privateKey = new Buffer(config.salt.key)
   try {
+    // Adding this here as an example of how to connect to the DB
+    const knex = connectToDatabase()
```

Comment:
Instead of returning the raw knex object, it would nice to raise the abstraction to the caller. We can wrap the database with a new class that provides high level operations and provide a strongly typed class for PhoneNumber to perform any validation. Ex.
interface PhoneNumberDatabase {
  hasPhoneNumber(number: PhoneNumber): Promise<boolean>
  getLookupsByPhoneNumber(number: PhoneNumber): Promise<number>
  incrementPhoneNumberLookup(number: PhoneNumber, amount: number)
  ...
}

This will ensure that state management is not spread out throughout the code and allow us to:
1. Provide an interface that can be extended in the future to other state store types (ex. in mem cache)
2. Make it easier to reason about the state management (ex. which race conditions do we need to worry about)
3. Add/remove properties to the state object in a single place

Reply:
I do think this would be nice -- perhaps there is existing tooling that integrates with knex which allows definition of the database schema in JSON with parser to generate the migrations and js wrappers.  otherwise could just create the wrappers manually

Reply:
I totally agree and did intend to create a `model` layer that abstracts away knex. Try not to overthink these changes, I'm just setting up the skeleton for the db connection :)